File: VJ/S_Uneven_Numbers.py
N = int(input())
n = 1
rest = N
cnt = 0
while (n <= N):
    l = 0
    tn = n
    while tn > 0:
        l += 1
        tn //= 10
    if l % 2 == 1:
        cnt += min(9 * n, rest)
    rest = max(0, rest - 9 * n)
    n *= 10
print(cnt)

File: VJ/S_Uneven_Numbers.cs
using System;

internal class Program
{
    private static int getd(int n)
    {
        int ans = 0;
        while (n > 0)
        {
            ans++;
            n /= 10;
        }
        return ans;
    }

    static void Main()
    {
        int N = int.Parse(Console.ReadLine());
        int n = 1, rest = N, cnt = 0;
        while (n <= N)
        {
            if (getd(n) % 2 == 1)
            {
                cnt += Math.Min(9 * n, rest);
            }
            rest -= 9 * n;
            n *= 10;
        }
        Console.WriteLine(cnt);
    }
}